kong/plugins/prometheus/exporter.lua

```diff
@@ -66,44 +66,50 @@ local function init()
+    memory_stats.shm_capacity:set(value.capacity, { node_id, shm_name, kong_subsystem })
   end
 
   metrics.memory_stats = memory_stats
 
   -- per service/route
   if kong_subsystem == "http" then
-    metrics.status = prometheus:counter("http_status",
-                                        "HTTP status codes per service/route in Kong",
-                                        {"service", "route", "code"})
+    metrics.status = prometheus:counter("http_requests_total",
+                                        "HTTP status codes per consumer/service/route in Kong",
+                                        {"service", "route", "code", "source", "consumer"})
   else
     metrics.status = prometheus:counter("stream_status",
-                                        "Stream status codes per service/route in Kong",
-                                        {"service", "route", "code"})
+                                        "Stream status codes per consumer/service/route in Kong",
+                                        {"service", "route", "code", "source", "consumer"})
   end
-  metrics.latency = prometheus:histogram("latency",
-                                         "Latency added by Kong, total " ..
-                                         "request time and upstream latency " ..
-                                         "for each service/route in Kong",
-                                         {"service", "route", "type"},
-                                         DEFAULT_BUCKETS) -- TODO make this configurable
-  metrics.bandwidth = prometheus:counter("bandwidth",
-                                         "Total bandwidth in bytes " ..
-                                         "consumed per service/route in Kong",
-                                         {"service", "route", "type"})
-  metrics.consumer_status = prometheus:counter("http_consumer_status",
-                                          "HTTP status codes for customer per service/route in Kong",
-                                          {"service", "route", "code", "consumer"})
+  metrics.kong_latency = prometheus:histogram("kong_latency_ms",
+                                              "Latency added by Kong and enabled plugins " ..
+                                              "for each service/route in Kong",
+                                              {"service", "route"},
+                                              KONG_LATENCY_BUCKETS)
+  metrics.upstream_latency = prometheus:histogram("upstream_latency_ms",
+                                                  "Latency added by upstream response " ..
+                                                  "for each service/route in Kong",
+                                                  {"service", "route"},
+                                                  UPSTREAM_LATENCY_BUCKETS)
+  metrics.total_latency = prometheus:histogram("total_request_latency_ms",
```

Comment:
`total_session_latency_ms`?

Reply:
🤔 my feeling is to align the name with its friends as possible, so if we have `total_` here, one might confuse with "is kong_latency` not "total kong_latency" and what's the other parts of `kong_latency`? usually we don't put `total_` in the latency names and use it for accumulative counters mostly. So I would suggest to just go with `session_latency_ms`. how do you think?

Reply:
updated to `session_latency_ms`

Reply:
-1 on session.
Sessions have a much different connotation than requests in our world. A session usually has multiple requests.
total_request_latency_ms is not the best name but it is better than session_latency_ms.

Reply:
okay. well. @sichvoge Please advise. Mr. Product, help us :)

Reply:
@hbagdi `session_latency` is for stream subsystem only, though there we can still talk about for example `TCP requests`, but that's less common compared to `TCP session`. It comes from the nginx `session_time` variable (http://nginx.org/en/docs/stream/ngx_stream_log_module.html).

Reply:
Do we use one metric for both stream and http subsystem?

Reply:
Oh right... maybe we should seperate the two then 👍 Maybe `request_latency_ms` for http and `session_latency_ms` for stream, it feels ambigious to use one for the other.

Reply:
tcp_session_duration_ms?

Reply:
I like the idea separating them out. Let's do that. 

Therefore, if there aren't any significant objections, let's keep `request_latency_ms` for http and for streams, introduce a new metric and use Harry's suggestion: `tcp_session_duration_ms`.

@ryanmoore @hfukada I feel like we have consensus.

Reply:
Updated to split into `request_latency_ms` `tcp_session_duration_ms` for http/streams respectively.

Reply:
let's just say `session_duration_ms` instead of `tcp_session_duration_ms`, as it can also be UDP

Reply:
but... does the concept of `session` make sense in the context of `UDP`? communication in `UDP` does not require a connection to be established between the peers. What would actually be a `session` in a `UDP` stream?

Reply:
hey @fffonion do you still think it's important to change `session_duration_ms` to `tcp_session_duration_ms`?

Reply:
@casimiro in UDP that variable will be the time taken after a datagram is received and before a response is sent. if we must explictly name this `tcp_session_duration_ms`, then we should also have a `udp_xxx` (or drop this metrics when protocol is UDP). either way, this is more about consistency, using `tcp_session_duration_ms` to describe both TCP and UDP is not correct. I would suggest we go with what Nginx have (since this data is directly read from a nginx variable after all), and keep the name aligned with the original nginx variable `$session_time`.

Reply:
If we use the same metric for both, wouldn't we want to also have some label that identifies the protocol (either `udp` or `tcp`?

Reply:
changed the metric name to `session_duration_ms` given there isn't a simple way to distinguish between `UDP` stream and `TCP` stream.